BUILD.bazel

```diff
@@ -55,23 +56,39 @@ expand_template(
     template = "src/catch2/catch_user_config.hpp.in",
 )
 
+# Generated header library, modifies the include prefix to account for
+# generation path so that we can include <catch2/catch_user_config.hpp>
+# correctly.
+cc_library(
+    name = "catch2_generated",
+    hdrs = ["catch2/catch_user_config.hpp"],
+    copts = ["-std=c++14"],
```

Comment:
I would not add `copts` here or on any other `cc_library` rule - It will result in warnings on compilers where the option is called slightly different (Visual Studio 2022). Users should compile it using `bazel build --cxxopt=-std=c++20 //...` or whatever C++ Standard they like.

You can also create a `.bazelrc` file with a predefined configs, e. g.:
```
build:gcc9 --cxxopt=-std=c++2a
build:clang13 --cxxopt=-std=c++17
build:vs2019 --cxxopt=/std:c++17
build:vs2022 --cxxopt=/std:c++17
```

When on Windows with Visual Studio one can build catch2 via `bazel build --config=vs2022 //...`